File: Project euler/problem14.java
import java.math.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.io.*;

class Problem14{

	static int collatzProblem(int n){
		int length=0;
		while(n!=1){
			if(n%2==0)
				n=n/2;
			else
				n=3*n+1;
			length+=n;	
		}
		return length;
	}

	static int problem14(){
		int mayor=1;
		int cont=1;
		while(cont<1000000){
			if(collatzProblem(cont)>collatzProblem(mayor))
				mayor=cont;
			cont++;
		}
		return mayor;
	}
	
	public static void main(String[] args) throws IOException{	    
		System.out.println(problem14());
	}
}

File: Project euler/problem14.py
#------Problem 14------

def collatzProblem(n=0):
    cont=1
    while n!=1:
        if str(math.log(n,2)).endswith(".0"):
                cont+=math.log(n,2)
                break
        if n%2==0:
            n/=2
        else:
            n=3*n+1
        cont+=1
    return cont

def problem14():
    mayor=0
    num=0
    cont=2
    while cont<1000000:
        a=collatzProblem(cont)
        if a>mayor:
            mayor=a
            num=cont
        cont+=1
    return num
